hydra.py

```diff
@@ -0,0 +1,102 @@
+import requests
+
+
+requests.packages.urllib3.disable_warnings()
+__version__ = "0.1.0"
+
+
+class RequestException(ValueError):
+    def __init__(self, response):
+        super(RequestException, self).__init__(response)
+        self.response = response
+
+    def __str__(self):
+        return "failed {} -> {}\n{}".format(
+            self.response.url, self.response.status_code, self.response.text
+        )
+
+
+class Client(object):
+    def __init__(self, username, password):
+        self.url = "https://access.redhat.com/hydra/rest"
+        self.username = username
+        self.password = password
+
+    def _hydra(self, fn, endpoint, parameters=None, payload=None):
+        kwargs = {
+            "params": parameters,
+            "headers": {
+                "Accept": "application/json",
+                "User-Agent": "cluster-support-bot/{}".format(__version__),
+            },
+            "auth": (self.username, self.password),
+        }
+        if payload is not None:
+            kwargs["json"] = payload
+        response = fn("{}/{}".format(self.url, endpoint), **kwargs)
+
+        if response.status_code == 204:
+            return
+
+        if response.status_code != 200:
+            raise RequestException(response=response)
+
+        if not response.text:
+            return
+
+        return response.json()
+
+    def get_account_notes(self, account):
+        notes = self._hydra(
+            fn=requests.get, endpoint="accounts/{}/notes".format(account)
+        )
+        if notes is None:
+            return []
+
+        return [
+            {
+                "type": n["type"],
+                "subject": n["subject"],
+                "isRetired": n["isRetired"],
+                "note": n["body"],
+                "id": n["id"],
```

Comment:
Is there a benefit to filtering like this instead of:

```python
        return self._hydra(
            fn=requests.get, endpoint="accounts/{}/notes".format(account)
        ) or []
```
or some such?

Reply:
I am not sure. That was another hangover from previous maintainer's code, but before I change it, let me go see what the data structure looks like for multiple notes without doing something special.

Reply:
Looks like it returns a lot MORE data, which is not a bad thing, but it does seem to be excess. For instance, we really just need the `id`, `body`, `isRetired`, and `type` but the Full response includes the `numberOfDaysReviewPending`, `name` (no idea what this is, seems autogenerated), a huge extra body of data reporting `lastModifiedBy`, `lastModifiedById`, `lastModifiedDate`, `createdById`, and `createdDate`.

I don't think we will need this but I am concerned that we are just ham-fistedly just cutting it out. Should we just return the whole thing?

Reply:
In this API-wrapping method, I think we should just return the whole thing.  Do you expect problems if we do that?  We should only extract the fields we want to expose when consuming this output in `handle_detail`.

Reply:
That seems reasonable. No, not really any expected problems, more just slight concern over exposing too much I guess?
I suppose, on further thought, we can always come back to limit it further at a later time if we need to. I will clean that up shortly.